ipam/calico-ipam.go

```diff
@@ -222,24 +222,46 @@ func cmdDel(args *skel.CmdArgs) error {
 		return err
 	}
 
-	// Release the IP address by using the handle - which is workloadID.
+	// Release the IP address by using the handleID.
```

Comment:
@heschlie I find the ordering of the current logic to be pretty confusing, and a bit hard to follow. Roughly, we've got:

- calculate the handle ID
- calculate the workloadID
- attempt release using the handle ID
  - unset a bool if IP not assigned
- attempt release using the workload ID, check boolean, emit error based on value.
  - only log if does not exist, and the previous bool was unset. 

I'm not a huge fan of using a boolean here for the sole purpose of hiding a log.

What about a flow like this?

- calculate handle ID
- attempt release using handle ID
  - if does not exist error:
    - calculate workloadID
    - attempt release using the workload ID
      - if does not exist error, log warning and return nil
  - if another type of error:
    - log error and return it

We can encapsulate the legacy workloadID logic into a helper function to make this code a bit cleaner.

Reply:
Originally I was thinking that there might be a situation where both the `handleID` and `workloadId` would exist at the same time, so we would need to call both (though admittedly could have been written better) but I'm less certain of that now, so I think encapsulating the `workloadId` call  is the way to go.

Reply:
As discussed offline, let's still release both, but I think we can clean up the logic a bit.